src/java/org/apache/cassandra/index/sai/disk/format/OnDiskFormat.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.format;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.cassandra.db.ClusteringComparator;
+import org.apache.cassandra.db.lifecycle.LifecycleNewTracker;
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.SSTableContext;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.index.sai.disk.PerIndexWriter;
+import org.apache.cassandra.index.sai.disk.PerSSTableWriter;
+import org.apache.cassandra.index.sai.disk.PrimaryKeyMap;
+import org.apache.cassandra.index.sai.disk.SearchableIndex;
+import org.apache.cassandra.index.sai.memory.RowMapping;
+import org.apache.cassandra.index.sai.utils.PrimaryKeyFactory;
+import org.apache.cassandra.io.sstable.format.SSTableReader;
+
+/**
+ * An interface to the on-disk format of an index. This provides format agnostics methods
+ * to read and write an on-disk format.
+ *
+ * The methods on this interface can be logically mapped into the following groups
+ * based on their method parameters:
+ * <ul>
+ *     <li>Methods taking no parameters. These methods return static information about the
+ *     format. This can include static information about the per-sstable components</li>
+ *     <li>Methods taking just an {@link IndexContext}. These methods return static information
+ *     specific to the index. This can be information relating to the type of index being used</li>
+ *     <li>Methods taking an {@link IndexDescriptor}. These methods interact with the on-disk components or
```

Comment:
```suggestion
 *     <li>Methods taking an {@link IndexDescriptor}. These methods interact with the on-disk components,
```

Reply:
I know it's a bit verbose but I believe that `or` is correct here because it only does one of the potential things.